packages/next/server/dev/next-dev-server.ts

```diff
@@ -1418,6 +1428,11 @@ export default class DevServer extends Server {
         this.serverCSSManifest = super.getServerCSSManifest()
       }
       this.fontLoaderManifest = super.getFontLoaderManifest()
+      // before we re-evaluate a route module, we want to restore globals that might
+      // have been patched previously to their original state so that we don't
+      // patch on top of the previous patch, which would keep the context of the previous
+      // patched global in memory, creating a memory leak.
+      this.restorePatchedGlobals()
```

Comment:
Q: Should we change the patching strategy? like only patch once instead of restoring the fetch?